zeebe/gateway-rest/src/main/java/io/camunda/zeebe/gateway/rest/validator/RequestValidator.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Camunda License 1.0. You may not use this file
+ * except in compliance with the Camunda License 1.0.
+ */
+package io.camunda.zeebe.gateway.rest.validator;
+
+import static io.camunda.zeebe.gateway.rest.validator.ErrorMessages.ERROR_MESSAGE_DATE_PARSING;
+import static io.camunda.zeebe.gateway.rest.validator.ErrorMessages.ERROR_MESSAGE_EMPTY_UPDATE_CHANGESET;
+import static io.camunda.zeebe.protocol.record.RejectionType.INVALID_ARGUMENT;
+
+import io.camunda.zeebe.gateway.protocol.rest.Changeset;
+import io.camunda.zeebe.gateway.protocol.rest.UserTaskUpdateRequest;
+import io.camunda.zeebe.gateway.rest.RestErrorMapper;
+import java.time.ZonedDateTime;
+import java.time.format.DateTimeParseException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ProblemDetail;
+
+public final class RequestValidator {
+
+  public static Optional<ProblemDetail> validateUpdateRequest(
```

Comment:
🔧 This is rather specific to user tasks, actually. I know it looks general, but it isn't. At least not yet. Let's move it to the user task validator for now.